lib/src/node.c

```diff
@@ -612,24 +594,19 @@ TSNode ts_node_child_with_descendant(TSNode self, TSNode descendant) {
         return self;
       }
 
-      // Here we check the current self node and *all* of its zero-width token siblings that follow.
-      // If any of these nodes contain the target subnode, we return that node. Otherwise, we restore the node we started at
-      // for the loop condition, and that will continue with the next *non-zero-width* sibling.
-      TSNode old = self;
-      // While the next sibling is a zero-width token
-      while (ts_node_child_iterator_next_sibling_is_empty_adjacent(&iter, self)) {
-        TSNode current_node = ts_node_child_with_descendant(self, descendant);
-        // If the target child is in self, return it
-        if (!ts_node_is_null(current_node)) {
-          return current_node;
-        }
-        ts_node_child_iterator_next(&iter, &self);
-        if (self.id == descendant.id) {
-          return self;
+      // If the descendant is empty, we check whether `self` contains it or not.
+      if (is_empty && ts_node_child_count(self) > 0) {
```

Comment:
If I understand correctly, this will recursively check every descendant of every node before the `descendant` node, even if they can't possibly contain it.
I think it should also check that `ts_node_end_byte(self) >= end_byte` or `iter.position.bytes >= end_byte`

Reply:
Yeah that's a good point, thanks!